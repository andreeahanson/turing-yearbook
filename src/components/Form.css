.add-student-form {
  background: greenyellow;
  padding: 50px;
}

.add-new {
  height: 30px;
}