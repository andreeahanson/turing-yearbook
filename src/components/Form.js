import React, { Component } from 'react';
import './Form.css';


class Form extends Component {
  constructor() {
    super()
    this.state={
      name: '',
      quote: '',
      superlative: '',
      photo: 'https://r.hswstatic.com/w_907/gif/loggerhead-turtle.jpg',
    }
  }

  handleChange = event => {
    this.setState({[event.target.name] : event.target.value})
  }

  resetInputs = () => {
    this.setState({
      name: '',
      quote: '',
      superlative: '',
    })
  }

  submitNewStudent = event => {
    event.preventDefault();
    const newStudent = {id: Date.now(), ...this.state}
    
    this.props.addNewStudent(newStudent);
    this.resetInputs();
  }


  render () {
    return (
      <form className="add-student-form">
      <label>Add new student</label>
        <input className="add-new"
        type="text"
        name = "name"
        placeholder = "Name"
        value = {this.state.name}
        onChange = {event => this.handleChange(event)}
        />
        <input className="add-new"
        type="text"
        name = "quote"
        placeholder = "Quote"
        value = {this.state.quote}
        onChange = {event => this.handleChange(event)}
        />
        <input className="add-new"
        type="text"
        name = "superlative"
        placeholder = "Superlative"
        value = {this.state.superlative}
        onChange = {event => this.handleChange(event)}
        />
        {/* <input className="add-new"
        type="file"
        name = "photo"
        placeholder = "Upload photo link"
        accept = ".jpg, .png. .svg"
        value = {this.state.photo}
        onChange = {event => this.handleChange(event)}  
        /> */}
        <button onClick={event => this.submitNewStudent(event)}>Add</button>
      </form>
    )
  }
}

export default Form;