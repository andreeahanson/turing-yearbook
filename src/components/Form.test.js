import React from 'react';
import { shallow, mount } from 'enzyme';
import Form from './Form.js';

describe('Form', () => {
  it('should match the snapshot with all the data passed in correctly', () => {
    const wrapper = shallow(<Form 
    addNewStudent = {jest.fn()}
    />)

    expect(wrapper).toMatchSnapshot();
  })

  it('should call addNewStudent when button is clicked', () => {
    const submitNewStudentMock = jest.fn()
    const wrapper = mount(
    <Form
        addNewStudent={submitNewStudentMock}
    />)

    wrapper.find('.add-student-btn').simulate('click');
    expect(submitNewStudentMock).toBeCalled();
  })


})