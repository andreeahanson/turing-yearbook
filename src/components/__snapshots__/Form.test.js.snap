// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot with all the data passed in correctly 1`] = `
<form
  className="add-student-form"
>
  <label>
    Add new student
  </label>
  <input
    className="add-new"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="add-new"
    name="quote"
    onChange={[Function]}
    placeholder="Quote"
    type="text"
    value=""
  />
  <input
    className="add-new"
    name="superlative"
    onChange={[Function]}
    placeholder="Superlative"
    type="text"
    value=""
  />
  <button
    className="add-student-btn"
    onClick={[Function]}
  >
    Add
  </button>
</form>
`;
